core/tracestack.lua

```diff
@@ -61,20 +61,22 @@ local traceStack = {
       return afterFrame and "after " .. afterFrame:toLocationString() or nil
     end
     local info = top:toLocationString()
+    local locationFrame = top
     -- Not all stack traces have to carry location information.
     -- If the top stack trace does not carry it, find a frame which does.
     -- Then append it, because its information may be useful.
     if not top.line then
       for i = #traceStack - 1, 1, -1 do
         if traceStack[i].line then
           -- Found a frame which does carry some relevant information.
-          info = info .. " near " .. traceStack[i]:toLocationString(--[[skipFile=]] traceStack[i].file == top.file)
+          locationFrame = traceStack[i]
+          info = info .. " near " .. locationFrame:toLocationString(--[[skipFile=]] locationFrame.file == top.file)
           break
         end
       end
     end
     -- Print after, if it is in a relevant file
-    if afterFrame and (not base or afterFrame.file == base.file) then
+    if afterFrame and (not locationFrame or afterFrame.file == locationFrame.file) then
```

Comment:
Here `locationFrame` will never be anything other than `top`, and we already return out of the function if `top` is not set. Testing for `not x` would always be false, so the `or` construct here is more convoluted than in needs to be to get the job done.